deepchecks/nlp/utils/text_properties.py

```diff
@@ -374,28 +380,53 @@ def _predict(text: str, classifier, kind: str) -> float:
             # if even one sentence is too long, use part of the first one:
             if len(text_to_use) == 0:
                 text_to_use = cut_string(sentences[0], MAX_CHARS)
-            text = text_to_use
-
-        v = classifier(text)
-    except Exception:  # pylint: disable=broad-except
-        return np.nan
-    else:
-        if not v:
-            return np.nan
-        v = v[0]
-        if kind == 'toxicity':
-            return v['score']
-        elif kind == 'fluency':
-            label_value = 'LABEL_1'
-        elif kind == 'formality':
-            label_value = 'formal'
+            text_list_to_predict.append(text_to_use)
         else:
-            raise ValueError('Unsupported value for "kind" parameter')
-        return (
-            v['score']
-            if v['label'] == label_value
-            else 1 - v['score']
-        )
+            text_list_to_predict.append(text)
+
+    while reduced_batch_size >= 1:
+        try:
+            if reduced_batch_size == 1:
+                results = []
+                for text in text_list_to_predict:
+                    try:
+                        v = classifier(text)
+                        if not v:
+                            results.append(np.nan)
+                        elif kind == 'toxicity':
+                            results.append(v['score'])
+                        elif kind == 'fluency':
+                            results.append(v['score'] if v['label'] == 'LABEL_1' else 1 - v['score'])
+                        elif kind == 'formality':
+                            results.append(v['score'] if v['label'] == 'formal' else 1 - v['score'])
+                        else:
+                            raise ValueError('Unsupported value for "kind" parameter')
+                    except Exception:  # pylint: disable=broad-except
+                        results.append(np.nan)
+                return results  # Return the results if prediction is successful
+
+            v_list = classifier(text_list_to_predict, batch_size=reduced_batch_size)
+            results = []
+
+            for v in v_list:
+                if not v:
+                    results.append(np.nan)
+                elif kind == 'toxicity':
+                    results.append(v['score'])
+                elif kind == 'fluency':
+                    results.append(v['score'] if v['label'] == 'LABEL_1' else 1 - v['score'])
+                elif kind == 'formality':
+                    results.append(v['score'] if v['label'] == 'formal' else 1 - v['score'])
+                else:
+                    raise ValueError('Unsupported value for "kind" parameter')
+
+            return results  # Return the results if prediction is successful
+
+        except Exception:  # pylint: disable=broad-except
```

Comment:
seems that for a batch of size 32 that has one bad example you will always come the at the end to running everything sample by sample but after you run many unnecessary calculations

Reply:
Added a retry mechanism to avoid that